UI/update/sparkle-updater.mm

```diff
@@ -0,0 +1,100 @@
+#include "mac-update.hpp"
+
+#include <qaction.h>
+
+#import <Cocoa/Cocoa.h>
+#import <Sparkle/Sparkle.h>
+
+#ifdef __aarch64__
+#define BRANCH_SUFFIX "_arm64"
+#else
+#define BRANCH_SUFFIX "_x86"
+#endif
+
+@interface OBSUpdateDelegate : NSObject <SPUUpdaterDelegate> {
+}
+@property (copy) NSString *branch;
+@property (nonatomic) SPUStandardUpdaterController *updaterController;
+@end
+
+@implementation OBSUpdateDelegate {
+}
+
+@synthesize branch;
+
+- (nonnull NSSet<NSString *> *)allowedChannelsForUpdater:
+	(nonnull SPUUpdater *)updater
+{
+	NSString *branchName = [branch stringByAppendingString:@BRANCH_SUFFIX];
+	;
+	return [NSSet setWithObject:branchName];
+}
+
+- (void)observeCanCheckForUpdatesWithAction:(QAction *)action
+{
+	[_updaterController.updater
+		addObserver:self
+		 forKeyPath:NSStringFromSelector(@selector(canCheckForUpdates))
+		    options:(NSKeyValueObservingOptionInitial |
+			     NSKeyValueObservingOptionNew)
+		    context:(void *)action];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath
+		      ofObject:(id)object
+			change:(NSDictionary<NSKeyValueChangeKey, id> *)change
+		       context:(void *)context
+{
+	if ([keyPath isEqualToString:NSStringFromSelector(
+					     @selector(canCheckForUpdates))]) {
+		QAction *menuAction = (QAction *)context;
+		menuAction->setEnabled(
+			_updaterController.updater.canCheckForUpdates);
+	} else {
+		[super observeValueForKeyPath:keyPath
+				     ofObject:object
+				       change:change
+				      context:context];
+	}
+}
+
+- (void)dealloc
+{
+	@autoreleasepool {
+		[_updaterController.updater
+			removeObserver:self
+			    forKeyPath:NSStringFromSelector(
+					       @selector(canCheckForUpdates))];
+	}
+}
+
+@end
+
+// Creates and starts the updater. There's nothing else required.
+OBSSparkle::OBSSparkle(const char *branch, QAction *checkForUpdatesAction)
+{
+	@autoreleasepool {
```

Comment:
This file is basically mostly just copied from Sparkle's Qt example (it works!), not sure if we really need this here. https://sparkle-project.org/documentation/programmatic-setup/#:~:text=Create%20an%20Updater%20in%20Qt